### PR TITLE
Fix realtime logging

### LINKED ISSUE #1140: ClientOptions LogLevel not working
- Setting loglevel from clientOptions to `debug` is not working.
- LogHandler is not set in `AblyRealtime` class
- Link to internal discussion - https://ably-real-time.slack.com/archives/C0174D4AP4P/p1648733606858019

┆Issue is synchronized with this [Jira Uncategorised](https://ably.atlassian.net/browse/SDK-1712) by [Unito](https://www.unito.io)

### PR DESCRIPTION
**Pull request by sacOO7**
- Fixed #1140 
 
~- [x] Remove `isDebug` flag usage, since it introduces unnecessary complexity, and it's already handled as a part of~

### CODE REVIEW

**Comment by tomkirbygreen** [comment]
Before you create a PR from this please split the work contained herein into at least two _separate_ PRs. The first that limits itself to fixing the issue using the current architecture and the second, which we can talk about, that suggests non-local changes to the architecture.

**Comment by sacOO7** [comment]
As you can see, there are no architectural changes. I am just fixing a simple bug. You can always update **TODO's**. It's just a change I noticed as a part of fixing the bug. So, we have created a dedicated issue for the bug and the PR for the same.

File: src/IO.Ably.Shared/Realtime/Workflows/RealtimeWorkflow.cs [hunk 1]
```diff
@@ -200,6 +200,7 @@ internal async Task<IEnumerable<RealtimeCommand>> ProcessCommand(RealtimeCommand
             bool shouldLogCommand = !((command is EmptyCommand) || (command is ListCommand));
             try
             {
+                // TODO - Some tests starts failing once Logger.IsDebug check is removed.
                 if (Logger.IsDebug && shouldLogCommand)
```
Comment (sacOO7) [inline src/IO.Ably.Shared/Realtime/Workflows/RealtimeWorkflow.cs:204]: @marto83 When I remove `isDebug` check, some tests starts failing. That means, if someome enables debug logging -> current functionality will break. I think this particular check can cause application to fail. One of the tests that is failing is -> `WhenMessageReceivedAndFailsVcdiffDecoding_ShouldSendAttachProtocolMessageWithLastSuccessfulChannelSerial`

File: src/IO.Ably.Shared/AblyRest.cs [hunk 2]
```diff
@@ -135,10 +135,7 @@ private void InitializeAbly(IMobileDevice mobileDevice)
                 Logger = Options.Logger;
             }
 
-            if (Options.LogLevel.HasValue)
```
Comment (marto83) [inline src/IO.Ably.Shared/AblyRest.cs:138]: Is Options.LogLevel guaranteed to be set?

Comment (sacOO7) [reply src/IO.Ably.Shared/AblyRest.cs:138]: Yes, by default it's set to warning.

Comment (marto-stadion) [reply src/IO.Ably.Shared/AblyRest.cs:138]: 👍

File: src/IO.Ably.Shared/Realtime/Workflows/RealtimeWorkflow.cs (hunk 1 above)
Comment (sacOO7) [reply src/IO.Ably.Shared/Realtime/Workflows/RealtimeWorkflow.cs]: This bug/test has been fixed as a part of https://github.com/ably/ably-dotnet/pull/1157

File: src/IO.Ably.Shared/AblyRest.cs (hunk 2 above)
Comment (sacOO7) [reply src/IO.Ably.Shared/AblyRest.cs:138]: Thanks

**Comment by sacOO7** [comment]
@marto83 I have updated the PR as per discussion and bug fix in https://github.com/ably/ably-dotnet/pull/1157. You can quickly review and merge it.

**Review by marto83 — APPROVED** [review]
Looks good 👍🏼

**Review by tomkirbygreen — APPROVED** [review]

File: src/IO.Ably.Shared/MessageEncoders/MessageHandler.cs [hunk 3]
```diff
@@ -354,24 +351,21 @@ private static Result DecodePayloads(DecodingContext context, IEnumerable<IMessa
 
         private void LogResponse(AblyResponse response)
         {
-            if (Logger.IsDebug)
```
[the remark this answers is not in the document]
Comment (tomkirbygreen) [reply src/IO.Ably.Shared/MessageEncoders/MessageHandler.cs]: Thank you Martin!

**Pull request merged**